### DISCUSSION TITLE
Error render of Page failed

### DISCUSSION THREAD

**Discussion by AbdulSayyed** [hbstack/.github#59] [Issues and bugs]
I followed the instructions, but the npm run dev or hugo server does not work. I am using Go version 1.22.2 and Hugo extended  v0.119. I am getting following errors:

npm run dev                                                                 bash  17:30:41 

> dev
> hugo server --gc --enableGitInfo -D

hugo: downloading modules …
go: downloading github.com/hugomods/ascii v0.1.0
go: added github.com/hugomods/ascii v0.1.0
hugo: collected modules in 56286 ms
WARN  Module "github.com/hbstack/gallery" is not compatible with this Hugo version; run "hugo mod graph" for more information.
WARN  Module "github.com/hugomods/images" is not compatible with this Hugo version; run "hugo mod graph" for more information.
WARN  Module "github.com/hugomods/search" is not compatible with this Hugo version; run "hugo mod graph" for more information.
Watching for changes in /home/sayyed/data-drive/hugo/learnx/{archetypes,assets,content,layouts,package.json}
Watching for config changes in /home/sayyed/data-drive/hugo/learnx/config/_default, /home/sayyed/data-drive/hugo/learnx/config/development, /home/sayyed/data-drive/hugo/learnx/go.mod
Start building sites … 
hugo v0.119.0-b84644c008e0dc2c4b67bd69cccf87a41a03937e+extended linux/amd64 BuildDate=2023-09-24T15:20:17Z VendorInfo=brew

WARN  [icons] simple-icons's twitter doesn't exist.
ERROR render of "page" failed: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/hb@v0.12.2/layouts/_default/baseof.html:20:8": execute of template failed: template: docs/single.html:20:8: executing "docs/single.html" at <partial "hugopress/body-end" .>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/hugopress@v0.2.3/layouts/partials/hugopress/body-end.html:1:4": execute of template failed: template: partials/hugopress/body-end.html:1:4: executing "partials/hugopress/body-end.html" at <partial "hugopress/functions/render-hooks" (dict "Name" "body-end" "Page" .)>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/hugopress@v0.2.3/layouts/partials/hugopress/functions/render-hooks.html:20:8": execute of template failed: template: partials/hugopress/functions/render-hooks.html:20:8: executing "partials/hugopress/functions/render-hooks.html" at <partial .partial $ctx>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/search@v0.2.1/layouts/partials/hugopress/modules/hb-search/hooks/body-end.html:2:5": execute of template failed: template: partials/hugopress/modules/hb-search/hooks/body-end.html:2:5: executing "partials/hugopress/modules/hb-search/hooks/body-end.html" at <partial "search/assets/js" .Page>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/search@v0.13.4/layouts/partials/search/assets/js.html:1:11": execute of template failed: template: partials/search/assets/js.html:1:11: executing "partials/search/assets/js.html" at <partialCached "search/assets/js-resource" .>: error calling partialCached: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/search@v0.13.4/layouts/partials/search/assets/js-resource.html:84:44": execute of template failed: template: partials/search/assets/js-resource.html:84:44: executing "partials/search/assets/js-resource.html" at <hugo>: can't evaluate field IsMultihost in type interface {}
ERROR render of "page" failed: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/hb@v0.12.2/layouts/_default/baseof.html:20:8": execute of template failed: template: _default/landing.html:20:8: executing "_default/landing.html" at <partial "hugopress/body-end" .>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/hugopress@v0.2.3/layouts/partials/hugopress/body-end.html:1:4": execute of template failed: template: partials/hugopress/body-end.html:1:4: executing "partials/hugopress/body-end.html" at <partial "hugopress/functions/render-hooks" (dict "Name" "body-end" "Page" .)>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/hugopress@v0.2.3/layouts/partials/hugopress/functions/render-hooks.html:20:8": execute of template failed: template: partials/hugopress/functions/render-hooks.html:20:8: executing "partials/hugopress/functions/render-hooks.html" at <partial .partial $ctx>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/search@v0.2.1/layouts/partials/hugopress/modules/hb-search/hooks/body-end.html:2:5": execute of template failed: template: partials/hugopress/modules/hb-search/hooks/body-end.html:2:5: executing "partials/hugopress/modules/hb-search/hooks/body-end.html" at <partial "search/assets/js" .Page>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/search@v0.13.4/layouts/partials/search/assets/js.html:1:11": execute of template failed: template: partials/search/assets/js.html:1:11: executing "partials/search/assets/js.html" at <partialCached "search/assets/js-resource" .>: error calling partialCached: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/search@v0.13.4/layouts/partials/search/assets/js-resource.html:84:44": execute of template failed: template: partials/search/assets/js-resource.html:84:44: executing "partials/search/assets/js-resource.html" at <hugo>: can't evaluate field IsMultihost in type interface {}
Built in 123729 ms
Error: error building site: render: failed to render pages: render of "page" failed: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/hb@v0.12.2/layouts/_default/baseof.html:20:8": execute of template failed: template: docs/single.html:20:8: executing "docs/single.html" at <partial "hugopress/body-end" .>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/hugopress@v0.2.3/layouts/partials/hugopress/body-end.html:1:4": execute of template failed: template: partials/hugopress/body-end.html:1:4: executing "partials/hugopress/body-end.html" at <partial "hugopress/functions/render-hooks" (dict "Name" "body-end" "Page" .)>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/hugopress@v0.2.3/layouts/partials/hugopress/functions/render-hooks.html:20:8": execute of template failed: template: partials/hugopress/functions/render-hooks.html:20:8: executing "partials/hugopress/functions/render-hooks.html" at <partial .partial $ctx>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hbstack/search@v0.2.1/layouts/partials/hugopress/modules/hb-search/hooks/body-end.html:2:5": execute of template failed: template: partials/hugopress/modules/hb-search/hooks/body-end.html:2:5: executing "partials/hugopress/modules/hb-search/hooks/body-end.html" at <partial "search/assets/js" .Page>: error calling partial: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/search@v0.13.4/layouts/partials/search/assets/js.html:1:11": execute of template failed: template: partials/search/assets/js.html:1:11: executing "partials/search/assets/js.html" at <partialCached "search/assets/js-resource" .>: error calling partialCached: "/home/sayyed/.cache/hugo_cache/modules/filecache/modules/pkg/mod/github.com/hugomods/search@v0.13.4/layouts/partials/search/assets/js-resource.html:84:44": execute of template failed: template: partials/search/assets/js-resource.html:84:44: executing "partials/search/assets/js-resource.html" at <hugo>: can't evaluate field IsMultihost in type interface {}

**Comment by razonyang** [comment]
Hi, thanks for getting interested in our themes, from the logs you posted above, you're using a older Hugo version which unsupported by some modules, upgrade Hugo version should work, I'd recommend using latest version of Hugo, at least `0.124.0`.

```
...
is not compatible with this Hugo version
...
```

**Comment by AbdulSayyed** [comment]
Both go and hugo updated to the latest issue is same on windows and linux. Only thing works is all three modules which are giving warning are set to earlier versions.

**Reply by razonyang** [reply]
What error do you get after upgrading Hugo and re-running Hugo server? As to the log, you're running hugo v0.119.0, however latest Hugo version is [`0.126.1`](https://github.com/gohugoio/hugo/releases).

**Comment by AbdulSayyed** [comment]
On Ubuntu it is working with latest update, both go and hugo must be updated. But on windows the error is following:
@books ➜ K:\..\learnx (1.22.3) git(main) 421ms go version
go version go1.22.3 windows/amd64                                                                                                                                                                                                                                     _default/gallery.html:4:8: executing "_def
@books ➜ K:\..\learnx (1.22.3) git(main) 174ms hugo version                                                                                                                                                                                                          :4": execute of t partials/hugopress/head-
hugo v0.122.0-b9a03bd59d5f71a529acb3e33f995e0ef332b3aa+extended windows/amd64 BuildDate=2024-01-26T15:54:24Z VendorInfo=gohugoio                                                                                                                                      modules\filecache2.1\layouts\partials\hugo
@books ➜ K:\..\learnx (1.22.3) git(main) 685ms npm run dev                                                                                                                                                                                                           ml" at <partial .che\modules\filecache\mod
                                                                                                                                                                                                                                                                      ailed: template: odules/seo-open-graph/hoo
> dev                                                                                                                                                                                                                                                                 rnal/opengraph.htlate failed: template: pa
> hugo server --gc --disableFastRender --enableGitInfo -D -p 1133                                                                                                                                                                                                      calling first: c

hugo: downloading modules …
hugo: collected modules in 5018 ms
Watching for changes in K:\ABC-Docs\@Archives\2024\learnx\{archetypes,assets,content,layouts,package.json}
Watching for config changes in K:\ABC-Docs\@Archives\2024\learnx\config\_default, K:\ABC-Docs\@Archives\2024\learnx\config\development, K:\ABC-Docs\@Archives\2024\learnx\go.mod
Start building sites … 
hugo v0.122.0-b9a03bd59d5f71a529acb3e33f995e0ef332b3aa+extended windows/amd64 BuildDate=2024-01-26T15:54:24Z VendorInfo=gohugoio

ERROR render of "page" failed: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hbstack\hb@v0.7.1\layouts\_default\baseof.html:4:8": execute of template failed: template: _default/landing.html:4:8: executing "_default/landing.html" at <partial "hugopress/head-begin" .>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\head-begin.html:1:4": execute of template failed: template: partials/hugopress/head-begin.html:1:4: executing "partials/hugopress/head-begin.html" at <partial "hugopress/functions/render-hooks" (dict "Name" "head-begin" "Page" .)>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\functions\render-hooks.html:20:8": execute of template failed: template: partials/hugopress/functions/render-hooks.html:20:8: executing "partials/hugopress/functions/render-hooks.html" at <partial .partial $ctx>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\seo\modules\open-graph@v0.1.0\layouts\partials\hugopress\modules\seo-open-graph\hooks\head-begin.html:1:4": execute of template failed: template: partials/hugopress/modules/seo-open-graph/hooks/head-begin.html:1:4: executing "partials/hugopress/modules/seo-open-graph/hooks/head-begin.html" at <partial "seo/modules/open-graph/index" .Page>: error calling partial: execute of template failed: template: _internal/opengraph.html:6:15: executing "_internal/opengraph.html" at <partial "_funcs/get-page-images" .>: error calling partial: execute of template failed: template: partials/_funcs/get-page-images.html:18:21: executing "partials/_funcs/get-page-images.html" at <first 1 .>: error calling first: can't iterate over maps.Params
ERROR render of "home" failed: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hbstack\hb@v0.7.1\layouts\_default\baseof.html:4:8": execute of template failed: template: _default/index.html:4:8: executing "_default/index.html" at <partial "hugopress/head-begin" .>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\head-begin.html:1:4": execute of template failed: template: partials/hugopress/head-begin.html:1:4: executing "partials/hugopress/head-begin.html" at <partial "hugopress/functions/render-hooks" (dict "Name" "head-begin" "Page" .)>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\functions\render-hooks.html:20:8": execute of template failed: template: partials/hugopress/functions/render-hooks.html:20:8: executing "partials/hugopress/functions/render-hooks.html" at 
<partial .partial $ctx>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\seo\modules\open-graph@v0.1.0\layouts\partials\hugopress\modules\seo-open-graph\hooks\head-begin.html:1:4": execute of 
template failed: template: partials/hugopress/modules/seo-open-graph/hooks/head-begin.html:1:4: executing "partials/hugopress/modules/seo-open-graph/hooks/head-begin.html" at <partial "seo/modules/open-graph/index" .Page>: error calling partial: execute of template failed: template: _internal/opengraph.html:6:15: executing "_internal/opengraph.html" at <partial "_funcs/get-page-images" .>: error calling partial: execute of template failed: template: partials/_funcs/get-page-images.html:18:21: executing "partials/_funcs/get-page-images.html" at <first 1 .>: error calling first: can't iterate over maps.Params
ERROR render of "section" failed: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hbstack\hb@v0.7.1\layouts\_default\baseof.html:4:8": execute of template failed: template: archives/list.html:4:8: executing "archives/list.html" at <partial "hugopress/head-begin" .>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\head-begin.html:1:4": execute of template failed: template: partials/hugopress/head-begin.html:1:4: executing "partials/hugopress/head-begin.html" at <partial "hugopress/functions/render-hooks" (dict "Name" "head-begin" "Page" .)>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\functions\render-hooks.html:20:8": execute of template failed: template: partials/hugopress/functions/render-hooks.html:20:8: executing "partials/hugopress/functions/render-hooks.html" at <partial .partial $ctx>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\seo\modules\open-graph@v0.1.0\layouts\partials\hugopress\modules\seo-open-graph\hooks\head-begin.html:1:4": execute of template failed: template: partials/hugopress/modules/seo-open-graph/hooks/head-begin.html:1:4: executing "partials/hugopress/modules/seo-open-graph/hooks/head-begin.html" at <partial "seo/modules/open-graph/index" .Page>: error calling partial: execute of template failed: template: _internal/opengraph.html:6:15: executing "_internal/opengraph.html" at <partial "_funcs/get-page-images" .>: error calling partial: execute of template failed: template: partials/_funcs/get-page-images.html:18:21: executing "partials/_funcs/get-page-images.html" at <first 1 .>: error calling first: can't iterate over maps.Params
ERROR render of "page" failed: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hbstack\hb@v0.7.1\layouts\_default\baseof.html:4:8": execute of template failed: template: _default/gallery.html:4:8: executing "_default/gallery.html" at <partial "hugopress/head-begin" .>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\head-begin.html:1:4": execute of template failed: template: partials/hugopress/head-begin.html:1:4: executing "partials/hugopress/head-begin.html" at <partial "hugopress/functions/render-hooks" (dict "Name" "head-begin" "Page" .)>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\functions\render-hooks.html:20:8": execute of template failed: template: partials/hugopress/functions/render-hooks.html:20:8: executing "partials/hugopress/functions/render-hooks.html" at <partial .partial $ctx>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\seo\modules\open-graph@v0.1.0\layouts\partials\hugopress\modules\seo-open-graph\hooks\head-begin.html:1:4": execute of template failed: template: partials/hugopress/modules/seo-open-graph/hooks/head-begin.html:1:4: executing "partials/hugopress/modules/seo-open-graph/hooks/head-begin.html" at <partial "seo/modules/open-graph/index" .Page>: error calling partial: execute of template failed: template: _internal/opengraph.html:6:15: executing "_internal/opengraph.html" at <partial "_funcs/get-page-images" .>: error calling partial: execute of template failed: template: partials/_funcs/get-page-images.html:18:21: executing "partials/_funcs/get-page-images.html" at <first 1 .>: error calling first: can't iterate over maps.Params
Built in 29343 ms
Error: error building site: render: failed to render pages: render of "section" failed: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hbstack\hb@v0.7.1\layouts\_default\baseof.html:4:8": execute of template failed: template: archives/list.html:4:8: executing "archives/list.html" at <partial "hugopress/head-begin" .>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\head-begin.html:1:4": execute of template failed: template: partials/hugopress/head-begin.html:1:4: executing "partials/hugopress/head-begin.html" at <partial "hugopress/functions/render-hooks" (dict "Name" "head-begin" "Page" .)>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\hugopress@v0.2.1\layouts\partials\hugopress\functions\render-hooks.html:20:8": execute of template failed: template: partials/hugopress/functions/render-hooks.html:20:8: executing "partials/hugopress/functions/render-hooks.html" at <partial .partial $ctx>: error calling partial: "C:\Users\books\AppData\Local\hugo_cache\modules\filecache\modules\pkg\mod\github.com\hugomods\seo\modules\open-graph@v0.1.0\layouts\partials\hugopress\modules\seo-open-graph\hooks\head-begin.html:1:4": execute of template failed: template: partials/hugopress/modules/seo-open-graph/hooks/head-begin.html:1:4: executing "partials/hugopress/modules/seo-open-graph/hooks/head-begin.html" at <partial "seo/modules/open-graph/index" .Page>: error calling partial: execute of template failed: template: _internal/opengraph.html:6:15: executing "_internal/opengraph.html" at <partial "_funcs/get-page-images" .>: error calling partial: execute of template failed: template: partials/_funcs/get-page-images.html:18:21: executing "partials/_funcs/get-page-images.html" at <first 1 .>: error calling first: can't iterate over maps.Params
@books ➜ K:\..\learnx (1.22.3) git(main) 49.914s

**Reply by razonyang** [reply]
May I have a access to your repo to reproduce the issue?

**Reply by razonyang** [reply]
I'd suggest upgrading all modules via `hugo mod get -u` (some module is quite old) and upgrade Hugo to latest (current is v0.122.0 on your local environment, latest is 0.126.1) if it's unavailable to share your repo.

**Comment by AbdulSayyed** [comment]
Thanks, I installed 0.126.1 extended on Windows, and it is working. With older down loads the trick hugo mod get -u worked thanks

**Reply by razonyang** [reply]
Glad it works.

**Comment by AbdulSayyed** [comment]
It is working on both Windows and Ubuntu
